frontend/src/js/constants.js

```diff
@@ -62,10 +62,30 @@ let iterateCardColors = function(index) {
     }
 };
 
+function insertObjectDifference(data, diff) {
+    if(diff === null || diff === undefined) {
+        return data;
+    }
+    let newData = {};
+    Object.keys(data).forEach(key => {
```

Comment:
look into [Array.prototype.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) maybe this could simplify this check. otherwise I am fine with the way this was solved

Reply:
Not sure if this would work since it is a recursive funktion